### ISSUE TITLE
NON_BIND_VARIABLE_SUFFIXES should only be used in PostgreSQL dialect

### ISSUE THREAD

**Issue by mingshewhe** [jOOQ/jOOQ#10345]
```
dslContext.fetch("select 60-40*(?-3)/(3-0)", 0).getValue(0, 0)
```
executed successfully on 3.9.6, but failed on 3.12.3, 
```
Caused by: java.sql.SQLException: No value specified for parameter 1
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:965)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:898)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:887)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:861)
	at com.mysql.jdbc.PreparedStatement.checkAllParametersSet(PreparedStatement.java:2211)
	at com.mysql.jdbc.PreparedStatement.fillSendPacket(PreparedStatement.java:2191)
	at com.mysql.jdbc.PreparedStatement.fillSendPacket(PreparedStatement.java:2121)
	at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1162)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:209)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:3928)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:294)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:368)
```

reason:
org.jooq.impl.Tools#renderAndBind
![image](https://user-images.githubusercontent.com/22971705/86359786-0b644300-bca4-11ea-838b-18c7e1b05b6a.png)
NON_BIND_VARIABLE_SUFFIXES should only be used in PostgreSQL dialect

### Versions:

- jOOQ:3.12.3
- ktolin: 1.2.10
- Database (include vendor): mysql5.7
- OS:
- JDBC Driver (include name if inofficial driver):5.1.46

**Comment by lukaseder** [comment]
Thanks for your suggestion. You're right, we could exclude this check from dialects that are not derived from PostgreSQL

**Comment by lukaseder** [comment]
The workaround is to simply add a space between `? -`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10347).

Thanks again for reporting, and investigating this in detail!
